src/Terminal/commands/grep.ts

```diff
@@ -185,16 +175,15 @@ class Args {
   private reduceToOptionsAndFiles(): string[] {
     const stripDash = (arg: string) => arg.slice(1);
     const argKeys = Object.keys(VALID_ARGS).map((k) => k as keyof Options);
-    const [allValidArgs, validFlagChars] = argKeys.reduce(
-      (ret: [readonly string[], string], key: keyof Options): [readonly string[], string] => {
-        const argStrings = VALID_ARGS[key];
-        return [
-          [...ret[0], ...argStrings.long, ...argStrings.short],
-          ret[1] + argStrings.short.map(stripDash).join(""),
-        ];
-      },
-      [[], ""],
-    );
+    const allValidArgs: string[] = [];
+
+    let validFlagChars = "";
+    for (const key of argKeys) {
+      const argString = VALID_ARGS[key];
+      allValidArgs.push(...[...argString.long, ...argString.short]);
```

Comment:
`push` can take multiple arguments, in fact that is what `...` is doing. So you can just do away the array here.

Reply:
Please don't apologise. :) My code can withstand many reviews!

I *think* it was broken as well haha. It "should" be good now...